libnest/src/package/identification.rs

```diff
@@ -1,7 +1,18 @@
 //! Package identification
 
-use crate::package::{CategoryName, PackageName, RepositoryName};
+use lazy_static::lazy_static;
+use regex::Regex;
 use semver::Version;
+use serde::de::Visitor;
+use serde::{Deserialize, Serialize};
+
+use crate::package::{CategoryName, PackageName, RepositoryName};
+
+lazy_static! {
+    static ref REGEX_PACKAGE_ID: Regex = Regex::new(
+        r"^((?P<repository>[a-z\-]+)::)?((?P<category>[a-z\-]+)/)?(?P<package>([a-z\-*]+[0-9]*))(#((?P<exact_version>[0-9.]+)|(?P<version_req>[<>=~^ ]*\s?[0-9.*]+)))?$"
```

Comment:
I think it would be safer to merge `exact_version` and `version_req` here, in a single capture group that matches pretty much everything after `#`, and let the rest be handled by `Version::parse()` or `VersionReq::parse()`

It is probably too much restrictive to consider numbers only at the end of a package name, tbh.